### ISSUE TITLE
Applying erased_serde::deserialize twice to same Deseralizer

### ISSUE THREAD

**Issue by voelklmichael** [dtolnay/erased-serde#38]
tl; dr; 
I'm confused by the &mut in erased_serde::deserialize.
May one use the deserializer afterwards?
Here is some code which panics:
```
#[derive(Debug, serde::Serialize, serde::Deserialize)]
struct D1 {}
#[derive(Debug, serde::Serialize, serde::Deserialize)]
struct D2 {}
trait T {}
impl T for D1 {}
impl T for D2 {}

use erased_serde::Deserializer;

fn main() {
    let d1 = D1 {};
    dbg!(&d1);
    let serialized = serde_json::to_string(&d1).unwrap();
    let mut deserializer = serde_json::Deserializer::from_str(&serialized);
    let mut deserializer = Deserializer::erase(&mut deserializer);
    dbg!(&erased_serde::deserialize::<D1>(&mut deserializer));
    dbg!(&erased_serde::deserialize::<D2>(&mut deserializer));
    dbg!(&erased_serde::deserialize::<D1>(&mut deserializer));
    dbg!(&serialized);
}
```

Hi, 
this is my first github issue, and this is more of a question.
here is some (unrelated) motivation.
I'm trying to write a plugin-system, and at the moment I'm checking if erased-serde can help.
To this end I'm runtime-loading some shared libraries, and from each I get a type implementing a shared trait.
Now I want to add a function to each library, which tries to deserialize the serialized trait into the libraries type.
Then, in the main executable, I wanted to iterate over my library-functions, feeding the same deseralizer into them.
Hence the example above.
Thank you,
Michael

**Comment by dtolnay** [comment]
Hi Michael, sorry that no one was able to provide guidance here. If this is still an issue, you could try taking this question to any of the resources shown in https://www.rust-lang.org/community.
